### ISSUE TITLE
I2I: Real Time config for amp-subscriptions-google

### ISSUE THREAD

**Issue by ratana** [ampproject/amphtml#24651]
Ability to assign different products to different SwG buttons / subscribe actions. For example, having one product attached to a paywall button and another attached to a dock button.

**Comment by jpettitt** [comment]
The main concern with this is that AMP documents can live in the cache for some time and we don't have a mechanism for marking sku's as no longer for sale.   What might be better would be to provide a remote config endpoint similar to the offered by amp-analytics.  This would allow a publisher to maintain a config file defining which sku to offer and even give them the option of making real time sku choices.

cc @dvoytenko

**Comment by dvoytenko** [comment]
I agree. Either we could have a config, or we could return the list of SKUs in the entitlements response as offers. We have some tools to render them into buttons and such.

**Comment by jpettitt** [comment]
Preliminary Design Doc
[AMP-Subscriptions-Google RTC.pdf](https://github.com/ampproject/amphtml/files/3688164/AMP-Subscriptions-Google.RTC.pdf)

**Comment by jpettitt** [comment]
I've been working on the implementation of this, it's mostly as the design doc with the exception of the data format.  The new format looks like this:

```JS
// Namespaced to potentially other platforms to use a similar model
"subscriptions.google.com": {
  // One sku, direct to pay flow
  "idOfButtonElement" : {
    "sku": "skuIdForDirectToPayflowCase"
  },
  // Carousel of offers
  "idOfAnotherButtonElement": {
    "carousel": {    // Object containing parms for a "showOffers" call see swg.js */
      "skus": [ "sku1", "sku2"]
    }
  }
}
```

By keying on the button ID it simplifies the code to just pass the element ID to the action the platform code.  In addition by having the `carousel` object be the params for the `showOffers()` it allows the publisher maximum flexibility in which offer to show and become extensible if we add other options to `showOffers()`.

**Comment by jpettitt** [comment]
@alin04 PTAL and let me know if you see any problems with updateSubscription on this data model.   It would be fairly easy to add the oldsku sku and proration model to this design to allow users to create dynamic update buttons in AMP.

**Comment by jpettitt** [comment]
Open question: what log events should this generate?

**Comment by alin04** [comment]
> @alin04 PTAL and let me know if you see any problems with updateSubscription on this data model. It would be fairly easy to add the oldsku sku and proration model to this design to allow users to create dynamic update buttons in AMP.

That should work just fine (adding oldsku and proration mode). My concern is support for contributions. The API supports:
subscribe()
showOffers()
contribute()
showContributionOptions()

How would this real-time config differentiation offers from contributions?

**Comment by jpettitt** [comment]
I think I need to put a type enum in the data structure to define contributions vs subscriptions

**Comment by jpettitt** [comment]
Decisions form internal design review: 

We will disable button until we know what they are going to do.  This means that if there is a remote config url the button will not be live until we receive the config.  On a timeout of config error we will either fallback to the server side amp offers carousel (default behavior) or leave the button disabled bset on a fallback mode flag.

Regarding login there is still an open question. v0 analytics events will simply log that a button was rendered but not try and log what sku's are associated with the button. This is consistent with existing behavior.  It's unclear if logging the sku is actually useful as different sku's will not have different ux at the button level.  It's probably better to allow the publisher to return a data value in the auth response that can be passed to analytics as this is where they should be delivering the UX for offers.

Separate feature request: allow automated shortcut to payflow on single sku carousels.  Detail TBD.

**Comment by jpettitt** [comment]
Implemented in #26223
